File: grimdb/sdk/swift/Sources/GrimlockerSDK/Models/VaultEntry.swift
import Foundation

public struct VaultEntry: Codable, Identifiable, Sendable {
    public let id: String
    public let title: String
    public let category: String
    public let fields: [String: String]?
    public let createdAt: Int64
    public let updatedAt: Int64

    enum CodingKeys: String, CodingKey {
        case id, title, category, fields
        case createdAt = "created_at"
        case updatedAt = "updated_at"
    }

    public func field(_ key: String) -> String { fields?[key] ?? "" }
}

File: grimdb/sdk/csharp/src/Models/VaultEntry.cs
using System.Text.Json.Serialization;

namespace Grimlocker.Models;

public record VaultEntry(
    [property: JsonPropertyName("id")]         string Id,
    [property: JsonPropertyName("title")]      string Title,
    [property: JsonPropertyName("category")]   string Category,
    [property: JsonPropertyName("created_at")] long CreatedAt,
    [property: JsonPropertyName("updated_at")] long UpdatedAt,
    [property: JsonPropertyName("fields")]     Dictionary<string, string>? Fields
);
